packages/react-native-reanimated/src/ReanimatedModule/js-reanimated/index.ts

```diff
@@ -159,7 +159,17 @@ const updatePropsDOM = (
 
   for (const key in domStyle) {
     if (isAnimatedProps) {
-      (component as HTMLElement).setAttribute(key, domStyle[key]);
+      // We need to explicitly set the 'text' property on input component because React Native's
+      // internal _valueTracker (https://github.com/facebook/react/blob/main/packages/react-dom-bindings/src/client/inputValueTracking.js)
+      // prevents updates when only modifying attributes.
+      if (
+        (component as HTMLElement).nodeName.toLowerCase() === 'input' &&
```

Comment:
nit: `nodeName` always contains the uppercase tag name of an HTML element
```suggestion
        (component as HTMLElement).nodeName === 'INPUT' &&
```